src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -527,8 +526,9 @@ public override BoundNode VisitSwitchExpression(BoundSwitchExpression node)
             for (int i = 0; i < numSwitchArms; i++)
             {
                 var nodeForSyntax = expressions[i];
+                var conversionOpt = node.SwitchArms[i].Value switch { BoundConversion c when c != nodeForSyntax => c, _ => null };
```

Comment:
>node.SwitchArms[i].Value switch { BoundConversion c when c != nodeForSyntax => c, _ => null } [](start = 36, length = 93)

Perhaps conditional operator rather than `switch`:
```
node.SwitchArms[i].Value as BoundConversion c &&
    c != nodeForSyntax ? c : null
```

Reply:
I had a ternary first, and I actually preferred the readability of the switch.

---
In reply to: [281691658](https://github.com/dotnet/roslyn/pull/35544#discussion_r281691658) [](ancestors = 281691658)